src/main/java/service/FruitBalanceService.java

```diff
@@ -0,0 +1,46 @@
+package service;
+
+import exceptions.BalanceException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitBalanceService implements BalanceService<String, Integer> {
+    private static final Map<String, Integer> fruitsBalance = new HashMap<>();
+
+    @Override
+    public void updateBalance(String fruit, Integer quantity) {
+        if (fruitsBalance.containsKey(fruit)) {
+            if (fruitsBalance.get(fruit) + quantity >= 0) {
```

Comment:
do we need this check?
let's just not put negative values to our map from the start